chain/ethereum/subscription.go

```diff
@@ -217,11 +217,13 @@ func (s *Subscription) subscribeHeaders(ctx context.Context) {
 		case header := <-headers:
 			//	Start a backfill when there are missing blocks
 			if s.latestBlockNumber != nil && header.Number.Uint64()-s.latestBlockNumber.Uint64() > 1 {
-				if backfillLogs, err := HistoricalEvents(ctx, s.client, s.addresses, s.latestBlockNumber.Uint64(), header.Number.Uint64()); err == nil {
-					for bfLog := range backfillLogs {
-						s.inLogs <- bfLog
+				go func() {
+					if backfillLogs, err := HistoricalEvents(ctx, s.client, s.addresses, s.latestBlockNumber.Uint64(), header.Number.Uint64()); err == nil {
```

Comment:
Conflicting due to latestBlockNumber is read and written from different goroutines at the same time

Reply:
it is written in the parent thread

Reply:
It is written in the parent goroutine and is read in in child goroutines. That will cause conflict. You should copy latestBlockNumber to other variable before passing it to the child goroutine.

Reply:
it is passed immediately to the backfill function as value. it is not shared resource. can you tell me a scenario where they conflict?

Reply:
No, it is not passed by value. You just access it by reference inside the child goroutine: `s.latestBlockNumber.Uint64()`

Reply:
new goroutine is created and it immediately calls `HistoricalEvents` which takes in uint64. So you are saying between creating the goroutine and calling `HistoricalEvents`, this property will be modified, while the only place modifies this property is in line 235?

Reply:
Correct, multi-threading is unpredictable, anything can happen. Whenever an object is accessed by multiple goroutines concurrently, it should be protected by a mutex.